### PR TITLE
rucredstash: 0.9.0 -> 0.9.2

### PR DESCRIPTION
**Pull request by psibi**
## Description of changes

Bumps the package to 0.9.2 and remove in-place Cargo.lock patch. Changelog details here: https://github.com/psibi/rucredstash/blob/master/CHANGELOG.md#v092

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/rucredstash/default.nix [hunk 1]
```diff
@@ -1,35 +1,28 @@
...
   doCheck = false;
 
-  # update Cargo.lock to work with openssl 3
-  postPatch = ''
-    ln -sf ${./Cargo.lock} Cargo.lock
-  '';
+  buildInputs =
+    [ lib.optional stdenv.isDarwin Security ];
+
+  cargoSha256 = "sha256-TYobVjjzrK3gprZcYyY98EvdASkq4urB+WiLlbJbwmk=";
```
Comment (wegank) [inline pkgs/tools/security/rucredstash/default.nix:21]:
```suggestion
  cargoHash = "sha256-TYobVjjzrK3gprZcYyY98EvdASkq4urB+WiLlbJbwmk=";
```

File: pkgs/tools/security/rucredstash/default.nix [hunk 2]
```diff
@@ -1,35 +1,28 @@
...
 
   meta = with lib; {
-    description = "Rust port for credstash. Manages credentials securely in AWS cloud";
+    description =
+      "Rust port for credstash. Manages credentials securely in AWS cloud.";
```
Comment (wegank) [inline pkgs/tools/security/rucredstash/default.nix:25]:
```suggestion
    description = "Utility for managing credentials securely in AWS cloud";
```

File: pkgs/tools/security/rucredstash/default.nix [hunk 3]
```diff
@@ -1,35 +1,28 @@
...
   # Disable tests since it requires network access and relies on the
   # presence of certain AWS infrastructure
   doCheck = false;
 
-  # update Cargo.lock to work with openssl 3
-  postPatch = ''
-    ln -sf ${./Cargo.lock} Cargo.lock
-  '';
+  buildInputs =
+    [ lib.optional stdenv.isDarwin Security ];
```
[the remark this answers is not in the document]
Comment (psibi) [reply pkgs/tools/security/rucredstash/default.nix]: Addressed, thanks!

File: pkgs/tools/security/rucredstash/default.nix (hunk 1 above)
Comment (psibi) [reply pkgs/tools/security/rucredstash/default.nix]: Addressed, thanks!

File: pkgs/tools/security/rucredstash/default.nix (hunk 2 above)
Comment (psibi) [reply pkgs/tools/security/rucredstash/default.nix]: Addressed, thanks!

**Pull request merged**